### PR TITLE
feat: cucumber tests (#7205)

### LINKED ISSUE #7205: Cypress tests to use Cucumber so more people can support
### What would you like to be added?

I am keen to add Cucumber to the Cypress testing, this will give an extra layer of abstraction.

Current
```js
   it('cluster', () => {
      NavbarPage.clickItem(NavbarPage.clusterItemId);
      NavbarPage.assertUrlContains('cluster');
    });
```

To
```gherkin
Scenario: Navigating to the cluster page
    Given I open "home" page
    When I click "#nav-cluster"
    Then I see "cluster" in the url
```

*note: more abstraction can be done with `#nav-cluster` etc, just an example*

### Why is this needed?

This will result in more people contributing to the tests and therefore improve the code coverage.

Q. I have started moving over a few examples, I can create a `draft` PR if useful?

### PR DESCRIPTION
**Pull request by eddiejaoude**
fixes #7205 

Move to using Cypress **with Cucumber**...

*note: I think it will be great for a live stream to get the community helping me move these tests over*

## Steps

1. Move existing tests from `.spec.ts` file to `.feature` file
1. Remove old spec test
1. Check off test in list below

## Test list

List of existing tests to move over.

### Cluster Items
- [x] cluster
- [x] cluster roles
- [x] cluster role binding
- [x] namespaces
- [x] nodes
- [x] persistentvolume
- [x] storageclass

Additional
- [ ] Network policy
- [ ] Role bindings
- [ ] Roles
- [ ] Service accounts

```bash
[1]        Spec                                              Tests  Passing  Failing  Pending  Skipped  
[1]   ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
[1]   │ ✔  nav/cluster.feature                      00:16        7        7        -        -        - │
[1]   └────────────────────────────────────────────────────────────────────────────────────────────────┘
[1]     ✔  All specs passed!                        00:16        7        7        -        -        -  
[1] 
```
### Workload and Namespace Items
- [ ] workload
- [ ] namespace selector

### Workloads Items
- [ ] workloads
- [ ] cronjob
- [ ] daemonset
- [ ] deployment
- [ ] job
- [ ] pod
- [ ] replicaset
- [ ] replicationcontroller
- [ ] statefulset

### Discovery and Load balancing Items
- [ ] discovery
- [ ] ingress
- [ ] service

### Config and Storage  Items
- [ ] config
- [ ] configmap
- [ ] persistentvolumeclaim
- [ ] secret

### Misc Items
- [ ] customresourcedefinition
- [ ] settings
- [ ] about

### Userpanel
- [ ] check default namespace
- [ ] collapses sidebar
- [ ] home logo click overview redirect check
- [ ] add resource
- [ ] search

Next steps: more tests can be added in a future PR to test additional functionality and improve code coverage

### CODE REVIEW

File: cypress/integration/nav/cluster.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Feature: Cluster page
+
+  Scenario: Navigating to the cluster page
+    Given I open "home" page
+    When I click "nav-cluster"
+    Then I see "cluster" in the url
+    And I see "Cluster" in the element "kd-breadcrumbs"
```
Comment (jamietanna) [inline cypress/integration/nav/cluster.feature:7]: We could make this more human readable, one of the main aims of Cucumber, by doing:

```suggestion
    And the breadcrumbs include "Cluster"
```

Comment (eddiejaoude) [reply cypress/integration/nav/cluster.feature:7]: Yes, but becomes less re-usable / flexible, so it is about finding a balance

File: cypress/integration/common/navigate.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      break;
+    case 'nav-persistentvolume':
+      element = '#nav-persistentvolume';
+      break;
+    case 'nav-storageclass':
+      element = '#nav-storageclass';
+      break;
+  }
+  cy.get(element).click();
+});
```
Comment (maciaszczykm) [inline cypress/integration/common/navigate.ts:41]: This whole method could be replaced by:

```
cy.get(`#${element}`).click();
```

**Comment by maciaszczykm** [comment]
In my opinion this adds additional complexity and a new layer to maintain. I think Cypress code is not that hard to read and if we want to bring improvements in that area we can prepare good interfaces that can be easily reused. I basically agree with what @floreks has said in https://github.com/kubernetes/dashboard/issues/7205 but couldn't respond at that time.

**Comment by maciaszczykm** [comment]
After short discussion with @floreks we agreed to close it for now. Especially with https://github.com/kubernetes/dashboard/pull/7047 being still in progress.

@eddiejaoude If you would like to contribute to E2E tests it would be really nice to add more of them once https://github.com/kubernetes/dashboard/pull/7047 will be done. You can find us on K8s Slack if you would like to discuss it.

**Comment by eddiejaoude** [comment]
Thanks @maciaszczykm , I am keen to add more tests and happy to continue to look after them. Sure I will try to join (I don't use Slack much though, more of a Discord person)

**Pull request closed without merging**